ui-library-add-call-recording/api/BlobStorage.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Microsoft.Extensions.Logging;
+
+namespace Contoso
+{
+    class BlobStorage
+    {
+
+        public BlobStorage(ILogger log, string connectionString, string containerName)
+        {
+            this.log = log;
+            var serviceClient = new BlobServiceClient(connectionString);
+            this.containerClient = serviceClient.GetBlobContainerClient(containerName);
+        }
+        public async Task<bool> UploadFileAsync(string blobName, Stream inStream)
+        {
+            if (!await this.containerClient.ExistsAsync())
+            {
+                throw new Exception("container not available");
+            }
+
+            var blobClient = this.containerClient.GetBlobClient(blobName);
+            if (await blobClient.ExistsAsync())
+            {
+                this.log.LogWarning($"Blob {blobName} already exists. Skipping upload.");
+                return false;
+            }
+
+            await blobClient.UploadAsync(inStream);
+            return true;
+        }
+
+        private ILogger log;
```

Comment:
In the past I have seen a majority of all variables stated at the top. Happy to get pushback here.

Reply:
I have absolutely (did I say absolutely) no idea of coding conventions in C#.

I'm used to writing code such that it's best to read top-down. Do you need to know what these private variables are to understand the key things in this class? No. So down they go :)

But happy to follow convention, because at the end of the day this is to be optimized for the readers.